templates/_helpers.tpl

```diff
@@ -33,6 +33,9 @@ HTTP|HTTPS Proxy Settings
   value: >-
     -D{{.Values.proxy.scheme}}.proxyHost={{ .Values.proxy.host }}
     -D{{.Values.proxy.scheme}}.proxyPort={{ .Values.proxy.port }}
+    {{- if not (empty .Values.proxy.nonProxyHosts) }}
```

Comment:
I think you _may_ be able to simplify this down to a `with` statement, though I can't recall if it handles empty lists.